_posts/2022-06-01-snitching-on-your-colleagues-using-cloud-magic.md

```diff
@@ -0,0 +1,332 @@
+
+<img class="p-image" src="{{ '/img/2022-06-01-snitching-on-your-colleagues-using-cloud-magic/ExampleStructure.png' | prepend: site.baseurl }}" class="image fit" style="margin:0px auto; max-width: 50%;">
+
+### Basic Tasks
+Following are the pipeline tasks used to test our java project, generate a test rapport and then build it using Maven.
+#### Maven Test
+To test our application we will be using JUnit because of the pre-existing support given by Azure. This will also generate a test rapport during each pipeline run based on the unit tests defined in our project.
+
+<img class="p-image" src="{{ '/img/2022-06-01-snitching-on-your-colleagues-using-cloud-magic/MavenTest.png' | prepend: site.baseurl }}" class="image fit" style="margin:0px auto; max-width: 50%;">
+
+#### Maven Build
+A standard task using the Maven goal of 'package' which returns a JAR file that can later be used by Docker.
+
+<img class="p-image" src="{{ '/img/2022-06-01-snitching-on-your-colleagues-using-cloud-magic/MavenBuild.png' | prepend: site.baseurl }}" class="image fit" style="margin:0px auto; max-width: 50%;">
+
+#### Copying And Creating An Artifact
+In order to use our build across multiple agents we need to create an artifact out of the build that then gets stored within the pipeline and can get called upon whenever we please.
+We do this by first copying our build to a directory on our agent that functions as the default staging directory for artifacts, then with another task we take that build and publish it to our pipeline storage.
+
+<img class="p-image" src="{{ '/img/2022-06-01-snitching-on-your-colleagues-using-cloud-magic/CopyAndPublish.png' | prepend: site.baseurl }}" class="image fit" style="margin:0px auto; max-width: 50%;">
+
+### Containerization
+Here we use the artifact we created during our last job to create an image and push it to Dockerhub.
+#### Fetching Our Artifact
+First we have to fetch the artifact that we uploaded to our pipeline and place it in the appropriate directory on our new agent.
+
+<img class="p-image" src="{{ '/img/2022-06-01-snitching-on-your-colleagues-using-cloud-magic/DownloadArtifact.png' | prepend: site.baseurl }}" class="image fit" style="margin:0px auto; max-width: 50%;">
+
+#### Creating An Image
+Then we use said artifact together with a Dockerfile that was previously placed within our java project to create and upload an image to Dockerhub.
+
+<img class="p-image" src="{{ '/img/2022-06-01-snitching-on-your-colleagues-using-cloud-magic/DockerTask.png' | prepend: site.baseurl }}" class="image fit" style="margin:0px auto; max-width: 50%;">
+
+### Setting Up A Cloud Database
+In this job we will use Terraform to set up an RDS (AWS) database based on Postgres for our containerized java application.
+Our Terraform files are stored in the java project under a folder called infrastructure and can be called upon directly, alternatively a remote Terraform can be specified if you want or need to split up your files.
```

Comment:
```suggestion
Our Terraform files are stored in the java project under a folder called infrastructure and can be called upon directly. 
Alternatively a remote Terraform module can be specified if you want to split up your files.
```

Rephrase the 2nd sentence. It's a bit weird now? Do you mean that you can reference external terraform modules in this terraform file? Or something else?